### PR TITLE
Hacktoberfest

### PR DESCRIPTION
**Pull request by gabanz**
## Added
- conversation/conversation/record-conversation.sh
  - add record and stop example in the same file
- account/topup-account-balance.sh
  - I'm not really sure about the syntax. Can someone verify it is correct?
- subaccounts/use-primary-account-balance.sh
- voice/voice-api/end-a-call.sh

### changed from beta (deprecated) to beta2
- conversation/event/list-events.sh 
- conversation/member/list-members.sh
- conversation/user/list-users.sh
### add option to stop playing
- voice/voice-api/play-audio-into-a-call.sh
- voice/voice-api/play-tts-into-a-call.sh

### CODE REVIEW

File: voice/voice-api/play-tts-into-a-call.sh [hunk 1]
```diff
@@ -6,3 +6,9 @@ curl -X PUT https://api.nexmo.com/v1/calls/$UUID/talk\
   -H "Authorization: Bearer "$JWT\
   -H "Content-Type: application/json"\
   -d '{"text": "Hello, hi there"}'
+
+sleep 5s
```
Comment (juliacodez) [inline voice/voice-api/play-tts-into-a-call.sh:10]: Why are you using `sleep`, then stopping the `talk` action?

File: voice/voice-api/play-audio-into-a-call.sh [hunk 2]
```diff
@@ -6,3 +6,9 @@ curl -X PUT https://api.nexmo.com/v1/calls/$UUID/stream \
   -H "Authorization: Bearer "$JWT\
   -H "Content-Type: application/json"\
   -d '{"stream_url": ["https://raw.githubusercontent.com/nexmo-community/ncco-examples/gh-pages/assets/welcome_to_nexmo.mp3"]}'
+
+sleep 5s
```
Comment (juliacodez) [inline voice/voice-api/play-audio-into-a-call.sh:10]: Could you please explain why you are using `sleep`, then stopping the `stream` action? The audio in this case is 2 seconds long, so by the time you get to the `DELETE` request, the audio stream will have already stopped.

File: voice/voice-api/play-tts-into-a-call.sh (hunk 1 above)
Comment (gabanz) [reply voice/voice-api/play-tts-into-a-call.sh:10]: it's an example of how to use cURL to stop the running talk action. I followed the sample in earmuff-a-call.sh and mute-a-call.sh 
i can remove it or put it into a separate shell script if that's more appropriate.

File: voice/voice-api/play-audio-into-a-call.sh (hunk 2 above)
Comment (gabanz) [reply voice/voice-api/play-audio-into-a-call.sh:10]: it's an example of how to use cURL to stop the running play action. I followed the sample in earmuff-a-call.sh and mute-a-call.sh 
i can remove it or put it into a separate shell script if that's more appropriate.

File: conversation/event/list-events.sh [hunk 3]
```diff
@@ -3,7 +3,7 @@
 source "../../config.sh"
 source "../../jwt.sh"
 
-curl "https://api.nexmo.com/beta/conversations/$CONVERSATION_ID/events" \
+curl "https://api.nexmo.com/beta2/conversations/$CONVERSATION_ID/events" \
```
Comment (tbedford) [inline conversation/event/list-events.sh:6]: Why changed to beta 2? The spec seems to specify v0.2.

File: conversation/member/list-members.sh [hunk 4]
```diff
@@ -3,7 +3,7 @@
 source "../../config.sh"
 source "../../jwt.sh"
 
-curl "https://api.nexmo.com/beta/conversations/$CONVERSATION_ID/members" \
+curl "https://api.nexmo.com/beta2/conversations/$CONVERSATION_ID/members" \
```
Comment (tbedford) [inline conversation/member/list-members.sh:6]: Why changed to beta 2? The spec seems to specify v0.2.

File: conversation/user/list-users.sh [hunk 5]
```diff
@@ -3,7 +3,7 @@
 source "../../config.sh"
 source "../../jwt.sh"
 
-curl "https://api.nexmo.com/beta/users" \
+curl "https://api.nexmo.com/beta2/users" \
```
Comment (tbedford) [inline conversation/user/list-users.sh:6]: Why changed to beta 2? The spec seems to specify v0.2.

File: voice/voice-api/play-tts-into-a-call.sh (hunk 1 above)
Comment (tbedford) [reply voice/voice-api/play-tts-into-a-call.sh:10]: We can't merge this specific change I'm afraid. We'd have to change the code snippet spec and then the code snippet for all languages and here there's no real benefit. The Code snippet is design to show playing TTS into a call and it currently does that.

File: voice/voice-api/play-audio-into-a-call.sh (hunk 2 above)
Comment (tbedford) [reply voice/voice-api/play-audio-into-a-call.sh:10]: We can't merge this specific change I'm afraid. We'd have to change the code snippet spec and then the code snippet for all languages and here there's no real benefit. The Code snippet is designed to show playing audio into a call and it currently does that.

**Comment by tbedford** [comment]
@gabanz - thanks for your contribution. I would have created one PR for each product. It makes each PR not only easier to test, but also to review and fix where needed. Also, we have https://github.com/Nexmo/code-snippet-specs which I'm not sure you're aware of? The problem is if we add a new feature into a code snippet, we'd have to change all code snippets to add the new functionality, and really we want the code snippets to really be as simple as possible. Those are my first thoughts. Currently I don't have access to the repo to add labels, do a proper review or merge, but I will come back to this PR once I do. Thanks again.

**Comment by lornajane** [comment]
Having reviewed the comments here, I'm going to close this pull request because we can't merge it as-is. @gabanz I really appreciate your input here, could you review the comments above and perhaps create one pull request per product? I think that makes it more likely that we can review and merge by pieces! Two things in particular that you have run into that are totally not your fault:
 - the code snippets specs are a private repo! I've got questions to ask about that but for now try to keep the functionality the same as the other code snippets in the same set (see developer.nexmo.com for where they are used)
 - we changed the URLs on the conversation API!!! Sorry!!

File: voice/voice-api/play-audio-into-a-call.sh (hunk 2 above)
Comment (gabanz) [reply voice/voice-api/play-audio-into-a-call.sh:10]: it's ok. i'll push a commit to undo the curl example calling the stop API. let me know if adding another bash file for that example makes more sense.

File: voice/voice-api/play-tts-into-a-call.sh (hunk 1 above)
Comment (gabanz) [reply voice/voice-api/play-tts-into-a-call.sh:10]: it's ok. i'll push a commit to undo the curl example calling the stop API. let me know if adding another bash file for that example makes more sense.

File: conversation/event/list-events.sh (hunk 3 above)
Comment (gabanz) [reply conversation/event/list-events.sh:6]: that's weird, i could have sworn i saw that beta2 in the API docs. but then i could be wrong, sorry about that. i'll undo this one.

File: conversation/member/list-members.sh (hunk 4 above)
Comment (gabanz) [reply conversation/member/list-members.sh:6]: that's weird, i could have sworn i saw that beta2 in the API docs. but then i could be wrong, sorry about that. i'll undo this one.

File: conversation/user/list-users.sh (hunk 5 above)
Comment (gabanz) [reply conversation/user/list-users.sh:6]: that's weird, i could have sworn i saw that beta2 in the API docs. but then i could be wrong, sorry about that. i'll undo this one.

File: voice/voice-api/play-audio-into-a-call.sh (hunk 2 above)
Comment (lornajane) [reply voice/voice-api/play-audio-into-a-call.sh:10]: I've closed this pull and request one pull request per product.

**Comment by gabanz** [comment]
thanks @lornajane, just saw your message. that makes sense. i thought there was a typo in the URL https://github.com/Nexmo/code-snippet-specs because it returned 404 status code to me.
Appreciate the time to review this. I'll create a new PR when I had the free time. 
cheers!

**Pull request closed without merging**